File: 프로그래머스/lv2/42587. 프린터/프린터.py
from collections import deque

def solution(priorities, location):
    answer = 0
    q = deque((v, idx) for idx,v in enumerate(priorities))
    while q:
        now = q.popleft()
        if q and now[0] < max(q)[0]:
            q.append(now)
        else:
            answer += 1
            if now[1] == location:
                break
    return answer

File: 프로그래머스/lv2/42587. 프린터/프린터.js
function solution(priorities, location) {
  var answer = 0;
  var q = [];
  for ([idx, prior] of Object.entries(priorities)) {
    q.push([prior, Number(idx)]);
  }
  priorities.sort((a, b) => b - a);

  while (q.length > 0) {
    var now = q.shift();
    if (now[0] >= Math.max(...priorities)) {
      answer++;
      if (now[1] === location) return answer;
      else priorities.shift();
    } else {
      q.push(now);
    }
  }
  return answer;
}